pkg/processor/runtime/rpc/connection/abstract.go

```diff
@@ -327,27 +345,65 @@ func (be *AbstractEventConnection) processItem(item interface{}, functionLogger
 		be.functionLogger = nil
 		return nil, errors.Wrapf(err, "Can't encode item: %+v", item)
 	}
-	processingResults, ok := <-be.resultChan
 
+	// if eventTimeout is 0, we wait for response without timeout
+	// if it is set, we wait either for response or the timeout
+	if be.connectionManager.GetConfig().eventTimeout == 0 {
+		processingResults, ok := <-be.resultChan
+		return be.postProcessEventRegularFlow(processingResults, ok)
+	} else {
+		return be.waitForResponseWithTimeout()
+	}
+}
+
+func (be *AbstractEventConnection) waitForResponseWithTimeout() (*result.BatchedResults, error) {
+	ticker := time.NewTicker(be.connectionManager.GetConfig().eventTimeout)
+	defer ticker.Stop()
+
+	select {
+	case processingResults, ok := <-be.resultChan:
+		return be.postProcessEventRegularFlow(processingResults, ok)
+	case <-ticker.C:
+		return be.postProcessEventOnTimeout()
+	}
+}
+
+func (be *AbstractEventConnection) postProcessEventRegularFlow(processingResults *result.BatchedResults, ok bool) (*result.BatchedResults, error) {
```

Comment:
`ok` is too broad, name it as what it actually checks (`isClientDisconnected` ?)